kv/util/runtime/value.go

```diff
@@ -37,24 +38,34 @@ var (
 
 // Value is a value that can be updated during runtime.
 type Value interface {
-	// Key is the key associated with value.
-	Key() string
-
 	// Watch starts watching for value updates.
 	Watch() error
 
 	// Unwatch stops watching for value updates.
 	Unwatch()
 }
 
-// UnmarshalFn unmarshals a kv value and extracts its payload.
-type UnmarshalFn func(value kv.Value) (interface{}, error)
+// Updatable can be updated.
+// TODO: Move to m3x/watch.
+type Updatable interface {
+	close.SimpleCloser
+
+	// C returns the notification channel for updates.
+	C() <-chan struct{}
+}
+
+// NewUpdatableFn creates an updatable.
+type NewUpdatableFn func() (Updatable, error)
+
+// GetFn returns the latest value.
+type GetFn func(updatable Updatable) (interface{}, error)
```

Comment:
Thought we were going to return a `Versionable` here?

Reply:
Because not all cases can be `Versionable`. For example just `boolean` and `int64` from `watch.Watch`

Reply:
Do we want to support a generic `watch.Watch` use case? I'd be happy to restrict it a little and only support cases where the value is always `Versionable` since it leads to much cleaner code. Is this something you need?

Reply:
I hope that we could leverage this runtime.Value to replace a bunch of Watch util functions in `kvutil` package. But I don't have to use it right now, could start with enforcing `Versionable` to begin with if you feel strongly.